2020-02-19/Nikhil/FindElementInSortedArray.java

```diff
@@ -0,0 +1,70 @@
+package arrays;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * @Author: Nikhil Reddy (nikoth)
+ * @Date: 19/02/20, Wed
+ **/
+
+/*
+Link to Problem Statement: https://github.com/vaskoz/dailycodingproblem-go/issues/553
+ */
+
+
+public class FindElementInSortedArray {
+
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        int noOfTests = sc.nextInt();
+
+        for(int test = 0; test < noOfTests; test++){
+
+            int length = sc.nextInt();
+
+            int[] numbers = new int[length];
+            for(int index = 0; index < length; index++){
+                numbers[index] = sc.nextInt();
+            }
+
+            int element = sc.nextInt();
+
+            Arrays.sort(numbers);
+
+            System.out.println(findMissingElement(numbers, element,0, length-1));
+        }
+    }
+
+    public static int findMissingElement(int[] numbers, int element, int left, int right){
+
+        if(left > right)
+            return -1;
+
+        int mid = getMid(left, right);
+
+        if(numbers[mid] == element)
+            return mid+1;
+
+        else if(element < numbers[mid])
+            return findMissingElement(numbers, element, left, mid-1);
+
+        return findMissingElement(numbers, element, mid+1, right);
+
+    }
+
+    public static int getMid(int left, int right){
+
+        int sum = left+right;
+        int count = 0;
+        while(sum -2 >= 0){
```

Comment:
You can decrease one subtraction operation by comparing like sum >=2

Reply:
with sum-2 >= 0 for sum = 8 comparsions are 5 at while.
with sum >= 0 for sum = 8 comparisions are 6 at while.

Reply:
Even the while loop will fall under O(logN) approach. 
If we take N = 9, then the while loop at max will start at left - 0 and right 8. 
log(9) = log(2^3 + 1 ) = log(2^3) + log(1) = 3+log(1) ~= 3. 
While loop also takes 4 iterations to get mid.


Even some details are missing. Few confusion are there. Need to discuss.

Reply:
log(a+b) != log(a)+log(b)

Reply:
I think it is not log(n). Complexity would be O(n/2) = ~O(n). If we want to calculate the mid for x, we are decreasing x  by 2 every time, it means we are doing x/2 iterations. This complexity is only for mid calculation.